framework/src/main/java/org/tron/core/net/message/handshake/HelloMessage.java

```diff
@@ -169,6 +170,22 @@ public boolean valid() {
       return false;
     }
 
+    int maxByteSize = 200;
+    ByteString address = this.helloMessage.getAddress();
+    if (!address.isEmpty() && address.toByteArray().length > maxByteSize) {
+      return false;
+    }
+
+    ByteString sig = this.helloMessage.getSignature();
+    if (!sig.isEmpty() && sig.toByteArray().length > maxByteSize) {
+      return false;
+    }
+
+    ByteString codeVersion = this.helloMessage.getCodeVersion();
+    if (!codeVersion.isEmpty() && codeVersion.toByteArray().length > maxByteSize) {
+      return false;
+    }
+
```

Comment:
The length judgment of address and version can be more specific.
If there are 3 multi-signatures, how long can the length be?

Reply:
this signature only come from SR's sig of timestamp in RelayService. it's 65 stablely, not related to multi-signature. You can read the code:

```
      String sig =
          TransactionCapsule.getBase64FromByteString(msg.getSignature());
      byte[] sigAddress = SignUtils.signatureToAddress(hash.getBytes(), sig,
          Args.getInstance().isECKeyCryptoEngine());
```